promql/engine.go

```diff
@@ -1332,6 +1348,7 @@ func (ev *evaluator) eval(expr parser.Expr) parser.Value {
 
 		res := newEv.eval(e.Expr)
 		ev.currentSamples = newEv.currentSamples
+		ev.updateSamples(ev.currentSamples, newEv.currentSamples)
```

Comment:
Might these have already been counted inside the subquery?

Also, not related to this PR but are we losing any other samples that were already in memory prior to the subquery?

Reply:
> Might these have already been counted inside the subquery?

Very true.

> Also, not related to this PR but are we losing any other samples that were already in memory prior to the subquery?

I don't think so. Since a subquery is just creating a new evaluator that handles evaluations and the `samplesStats` variable is a pointer so, if there were to be sample prior to the subquery, they would be already caught in that case.